framework/base/Security.php

```diff
@@ -521,14 +521,16 @@ public function generateRandomKey($length = 32)
             if ($urandomFile !== null) {
                 // $length could be large so read using a loop.
                 $key = '';
+                $bytesLeft = $length;
                 do {
-                    $buffer = fread($urandomFile, $length);
+                    $buffer = fread($urandomFile, min($bytesLeft, 8092));
```

Comment:
see "chunk size" in http://php.net/manual/en/function.fread.php
but we could also keep `$bytesLeft` here only.